R/shiny_prerendered.R

```diff
@@ -160,13 +160,20 @@ shiny_prerendered_html <- function(input_rmd, render_args) {
   # return html w/ dependencies
   html_with_deps <- shinyHTML_with_deps(rendered_html, dependencies)
 
-  # add placeholder for additional dependencies at the end of the <head> element
-  # using a template expansion expected by shiny (usually done in shiny:::renderPage)
-  sub('</head>',
-      paste0('\n', htmltools::htmlTemplate(text_ = "{{ headContent() }}"), '\n</head>'),
+  # The html template used to render the UI should contain the placeholder
+  # expected by shiny in `shiny:::renderPage()` which uses
+  # `htmltools::renderDocument`.
+  # If it is not present in the template, we add this placeholder at the end of
+  # the <head> element
+  if (!any(grepl(headContent <- "<!-- HEAD_CONTENT -->", html_with_deps, fixed = TRUE))) {
```

Comment:
Trying to keep the tag isolated to `htmltools`...
```suggestion
  headContent <- as.character(htmltools::htmlTemplate(text_ = "{{ headContent() }}"))
  if (!any(grepl(headContent, html_with_deps, fixed = TRUE))) {
```

Reply:
As discussed, we need to use the `<!-- HEAD_CONTENT -->` as raw in the HTML templates to not have them passed to `htmltools::htmlTemplates()`. This means there is no real value to use `headContent()` here as if it changes upstream we need to update the templates. 

I added a test to detect any change on this `headContent()` value in `htmltools::htmlTemplate()`. 

Note: `htmltools::renderDocument()` use the raw comment and documents it as being the token to use with this function. 

Thanks for the feedback!